x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateExtractTests.java

```diff
@@ -84,7 +84,12 @@ public static Iterable<Object[]> parameters() {
                         )
                         .withFoldingException(InvalidArgumentException.class, "invalid date field for []: not a unit")
                 )
-            )
+            ),
+            (v, p) -> switch (p) {
+                case 0 -> "string";
+                case 1 -> "datetime";
+                default -> "";
```

Comment:
should this throw on unknown?

Reply:
so we get to the default if the code that figures out which position has the bad argument comes up with a number higher than the number of arguments.  Like if it says "the fourth argument to + is bad".  That should really never happen, and I didn't put a lot of thought into what to do if it did, I just made the switch expression happy.

Reply:
I tend to make them happy by throwing if I don't expect it. I suppose in this case it's not a big difference.